### ISSUE TITLE
Got connection error when using 'truffle console'  to connect to the RPC endpoint run with 'ganache-cli'

### ISSUE THREAD

**Issue by zexxlin** [trufflesuite/truffle#3183]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Steps to Reproduce

1. run `ganache-cli`, listening on port 8545
2. run `truffle console` with minimal network configuration
>     development: {
>       host: '127.0.0.1',
>       port: 8545,
>       network_id: '*',
>       // websockets: true
>     }
3. Interactive shell shows up normally with `truffle(development)>` prompt.
4. Execute `await web3.eth.getAccounts()` and get CONNECTION ERROR.
 
I've tried to use MetaMask to connect to the local RPC endpoint, and everything works fine.

Also, if I add `websockets: true` option to the configuration to use WebsocketProvider instead, the issue's gone as well.

## Expected Behavior

RPC calls with default HTTPProvider should execute successfully.

## Actual Results

Receive uncaught error: CONNECTION ERROR: Couldn't connect to node http://127.0.0.1:8545.

## Environment

* Operating System: Arch Linux
* Ethereum client: ganache-cli v6.9.1 (ganache-core: 2.10.2)
* Truffle version (`truffle version`): v5.1.35
* node version (`node --version`): v12.18.2
* npm version (`npm --version`): v6.14.5

**Comment by CruzMolina** [comment]
Hey @zexxlin ! This also looks like it might be a `ganache-core` internal compatibility issue with Arch Linux.

Are you able to reproduce when running the `ganache-cli` Docker image? (`docker run -d -p 8545:8545 trufflesuite/ganache-cli:latest`)

**Comment by zexxlin** [comment]
Thank you @CruzMolina . I could still reproduce the issue when running `ganache-cli` Docker image. But downgrading to Node v10 resolves the issue. After a few experiments, I could confirm `truffle` has some compatibility issues with Node v12, at least on Arch Linux, while `ganache-cli` works fine in both cases.

**Comment by eggplantzzz** [comment]
Hey @zexxlin, I just want to confirm what the incompatibility is here. So you think there may be an incompatibility with Node 12 and running `truffle console`? We'll have to investigate this and see what's going on as we were not aware that this was an issue. Thanks for the report!

**Comment by zexxlin** [comment]
@eggplantzzz Exactly, but not sure if it's OS related or not, since I just witnessed it on Arch Linux.

**Comment by eggplantzzz** [comment]
I don't think this is related to Node 12. I am able to run `truffle console` and connect to an instance of Ganache running while using Node 12. There must be something else going on here. I am on a mac.

**Comment by zexxlin** [comment]
@eggplantzzz If I downgrade to Node 10 on Arch Linux, everything works fine. Considering it's working on your Mac, it's likely Node 12 and Arch don't work well together.

**Comment by eggplantzzz** [comment]
Ah yes sorry I guess we can't rule out that possibility yet. Let me see if I can get a colleague running Linux to try and reproduce your error. I'll let you know if I find anything!

**Comment by eggplantzzz** [comment]
Just for another data point; my colleague who runs Linux Mint was able to successfully connect using `truffle console`.

**Comment by dvargas92495** [comment]
I just hit this issue on Windows 10. Downgrading to node version `10.22.0` fixed it for me

**Comment by dvargas92495** [comment]
Even stranger, I just tested out this script on node 12:
```javascript
const createInterfaceAdapter = require("@truffle/interface-adapter")
  .createInterfaceAdapter;
const web3 = require("web3");

const adapter = createInterfaceAdapter({
  networkType: "ethereum",
  provider: new web3.providers.HttpProvider("http://127.0.0.1:7545"),
});

adapter
  .getBlock("latest")
  .then((g) => console.log(g))
  .catch((e) => console.error(e.message));
```

which to my understanding is what truffle console is doing under the hood based on the error message I get. This script runs successfully though. Is there anything weird that's happening when we're setting up the `Environment` that it fails within the truffle console?

**Comment by eggplantzzz** [comment]
It looks like what you are doing is standard enough. I was able to run your script successfully on a mac with both Node 12 and 14. We'll have to get a Windows setup to troubleshoot with.

**Comment by dvargas92495** [comment]
To be clear - the script runs successfully on my end too. What fails to connect is `truffle console` trying to connect to ganache in node 12. The script was just my attempt at trying to investigate what truffle console does under the hood to debug

**Comment by eggplantzzz** [comment]
Oh sorry! I misunderstood. I just confirmed that `truffle console` can connect to a ganache-cli network running locally on Node 12 and 14. Again, it would be better if I had a Windows machine to try and duplicate your error.

**Comment by eggplantzzz** [comment]
We think this should be fixed on the latest Truffle and Ganache v7 (which has been renamed to `ganache`). Closing this but if you are still having errors please open a new issue and we'll help troubleshoot. Thank you!

**Comment by BenjDrury** [comment]
I dont think it is solved - been having the same issue on 
Truffle v5.7.6 (core: 5.7.6)
Node v18.13.0

**Comment by suraj187104** [comment]
Do truffle develop before doing truffle migrate or console
this will solve your problem
